src/components/Footer.js

```diff
@@ -1,25 +1,78 @@
-import React from "react";
-import { View, Text, StyleSheet } from "react-native";
+import React, { useState } from "react";
+import { View, Text, StyleSheet, Modal, TouchableOpacity } from "react-native";
+import { useSelector, connect } from "react-redux";
+import Login from "./Login";
 import { AppLoading } from "expo";
 import { useFonts } from "@use-expo/font";
 import fonts from "./module/fonts";
 
-
-const Footer = () => {
+const Footer = ({ dispatch }) => {
   let [fontsLoaded] = useFonts(fonts);
+  const authenticated = useSelector((state) => state.authenticated);
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const showButton = authenticated ? (
+    <TouchableOpacity
+      testID={"Logoutbutton"}
+      onPress={() => {
```

Comment:
You need to destroy the session in the backend as well, so send a request there on logout press.

Reply:
Aha yes!